packages/vscode-extension/src/utilities/expoCli.ts

```diff
@@ -20,20 +20,35 @@ export function shouldUseExpoCLI() {
 
   const appRootFolder = getAppRootFolder();
   let hasExpoCLIInstalled = false,
-    hasExpoCommandsInScripts = false;
+    hasExpoCommandsInScripts = false,
+    hasExpoConfigInAppJson = false,
+    hasExpoConfigInAppConfigJs = false;
```

Comment:
These are very nitty:
1. this is a personal opinion, but I dislike multiple variable declarations in a single statement, I feel like splitting this would be more readable.
```suggestion
  let hasExpoCommandsInScripts = false;
  let hasExpoConfigInAppJson = false;
  let hasExpoConfigInAppConfigJs = false;
```
2. Personally, I'd rewrite this function with early returns, but that's a stylistic choice.
```
try {
  const hasExpoCliInstalled = ...
  if (!hasExpoCliInstalled) { return false }
} catch {}

try {
  const appJson = require(...)
  if ((appJson).includes("expo")) return true;
} catch {}

...

return false;
```

Reply:
And honestly, for 2. in this case, I think your approach is more readable -- you have the explicit condition in the return statement, while in mine you have to read the whole function to understand it.